### PR TITLE
Fix issue with empty character sets

### PR DESCRIPTION
**Pull request by olsaarik**
`GetFixedPrefix` was not working for empty predicates (probably since my iterative rewrite of it). This also enables a test with empty sets that was disabled for NonBackTracking.

### CODE REVIEW

**Review by veanes — APPROVED** [review]

File: src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexNode.cs [hunk 1]
```diff
@@ -1504,6 +1504,10 @@ private IEnumerable<S> GetPrefixSequence()
                     // A path didn't have a next set as supported by this algorithm
                     yield break;
                 }
+                if (!_builder._solver.IsSatisfiable(next))
+                {
+                        yield break;
```
Comment (stephentoub) [inline src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexNode.cs]: Nit:
```suggestion
                    yield break;
```

**Review by stephentoub — APPROVED** [review]
